components/cam/src/physics/cam/micro_p3_interface.F90

```diff
@@ -763,8 +763,13 @@ subroutine get_cloud_fraction(its,ite,kts,kte,ast,qc,qr,qitot,method, &
           rcldm(i,k) = cldm(i,k)
           IF (trim(method) == 'in_cloud') THEN
              IF (k /= ktop) THEN
+                ! If cloud mass exists, keep precip_frac = cldm (precipitation
+                ! only present in cloud). If not, use the max of cloud fraction
+                ! and fraction from the level above (precip is originating from
+                ! above in this case, but we need to use the max because P3 can't
+                ! handle rcldm < cldm correctly).
                 IF (qc(i,k) .lt. qsmall .and. qitot(i,k) .lt. qsmall) THEN
-                   rcldm(i,k) = rcldm(i,k+kdir)
+                   rcldm(i,k) = max(rcldm(i,k+kdir),rcldm(i,k))
```

Comment:
maybe add comment here: "max(rcldm above and rcldm for this layer) is taken here because code is incapable of handling rcldm<cldm for a given grid cell"

Reply:
Good suggestion, this has been added.